plugins/CoreVisualizations/Visualizations/Graph.php

```diff
@@ -135,26 +149,48 @@ public function afterAllFiltersAreApplied()
             ));
         }
 
-        // set default selectable columns, if none specified
-        $selectableColumns = $this->config->selectable_columns;
-        if (false === $selectableColumns) {
-            $selectableColumns = array('nb_visits', 'nb_actions', 'nb_uniq_visitors', 'nb_users');
-
-            if ($this->config->show_goals) {
-                $goalMetrics       = array('nb_conversions', 'revenue');
-                $selectableColumns = array_merge($selectableColumns, $goalMetrics);
-            }
-        }
-
         $transformed = array();
-        foreach ($selectableColumns as $column) {
+        foreach ($this->config->selectable_columns as $column) {
             $transformed[] = array(
                 'column'      => $column,
                 'translation' => @$this->config->translations[$column],
                 'displayed'   => in_array($column, $this->config->columns_to_display)
             );
         }
-
         $this->config->selectable_columns = $transformed;
     }
+
+    protected function generateSelectableColumns()
+    {
+        $selectableColumns = array('nb_visits', 'nb_actions');
```

Comment:
@katebutler not sure about the columns we add by default. Many reports won't have the columns `nb_actions`, `nb_conversions`, `revenue`, `nb_uniq_visitors ` and `nb_users`. Some reports might not even have the `nb_visits`. Did you copy this logic from somewhere?

Otherwise we could maybe simply set the first non-label column from the first row?

And if there is only `label` in the first row, we could maybe default to `nb_visits`?

Reply:
The existing logic was to just use all of them and hope for the best :smile:  but I have reworked it to use all columns except `label` if none of these are present in the dataset